lms/djangoapps/discussion/rest_api/api.py

```diff
@@ -80,11 +83,19 @@ class DiscussionTopic:
     Class for discussion topic structure
     """
 
-    def __init__(self, topic_id, name, thread_list_url, children=None):
+    def __init__(
+        self,
+        topic_id: str,
+        name: str,
+        thread_list_url: str,
+        children: Optional[List[DiscussionTopic]] = None,
+        thread_counts: Dict[str, int] = None,
+    ):
         self.id = topic_id  # pylint: disable=invalid-name
         self.name = name
         self.thread_list_url = thread_list_url
         self.children: List[DiscussionTopic] = children or []  # children are of same type i.e. DiscussionTopic
```

Comment:
Do we need a type hint `List[DiscussionTopic]` here after it is added in the args?

Reply:
I think it can be removed.